src/test/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTaskTest.java

```diff
@@ -377,6 +391,21 @@ public void testShouldNotThrowIfReporterDoesNotExist() {
     setUpTask();
   }
 
+  @Test
+  public void testShouldVerifyChangingTopic() {
+    props.put(FLUSH_SYNCHRONOUSLY_CONFIG, "false");
+    setUpTask();
+    String changedTopic = "routed-to-another-topic";
+    SinkRecord record = record(changedTopic, false, false, 0);
+    when(context.assignment()).thenReturn(Collections.singleton(new TopicPartition("original-topic-name", 1)));
+    try {
+      task.put(Collections.singletonList(record));
+    } catch (ConnectException e) {
+      assertEquals(String.format("Found a topic name '%s' that doesn't match assigned partitions."
```

Comment:
The test would pass if not exceptions are thrown. If we expect this exception to be thrown, it would be better to use `assertThatThrownBy` or similar.

Reply:
Good catch, switched to use `assertThrows`.